### PR TITLE
Add the subindexes plugin.

### PR DESCRIPTION
**Pull request by bwinton**
One of the things I missed from PyBlosxom was a sub-index listing all the posts underneath a subdirectory.
So I wrote this.  :)

Any suggestions about how to make it better would be greatly appreciated!

[Review on Reviewable](https://reviewable.io/reviews/getnikola/plugins/67)

### CODE REVIEW

**Comment by bwinton** [comment]
(Uh, that error doesn’t _seem_ like something I caused…)

File: v7/subindexes/subindexes.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
```
Comment (Kwpolska) [inline v7/subindexes/subindexes.py]: missing a license

(can be anything you like)

File: v7/subindexes/subindexes.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+from __future__ import unicode_literals
```
Comment (Kwpolska) [inline v7/subindexes/subindexes.py]: more like

```# Copyright © 2015 Blake Winton.```

**Comment by Kwpolska** [comment]
It looks like this plugin completely ignores archives.  You can easily have a blog which has indexes already there.

For example, my blog has https://chriswarrick.com/blog/2014/ and also https://chriswarrick.com/blog/2014/11/09/inbox-review/ via archives and posts/ structured mirrored in output.

**Comment by bwinton** [comment]
Neat.  How did you get https://chriswarrick.com/blog/2014/11/09/inbox-review/ to show up there?  Is the inbox-review post in a "2014/11/09" directory?

(Also, that seems to solve a slightly different problem than the one I was trying to solve, since I want the sub-indexes to show up, and https://chriswarrick.com/blog/2014/11/ doesn't show all the posts in 2014/11…  :)

**Comment by Kwpolska** [comment]
1. Yes, the path is `posts/2014/11/09/inbox-review.rst` (`/blog/` is handled by `POSTS`)
2. `2014/11` does not work because I didn’t enable `CREATE_MONTHLY_ARCHIVES`, which is overkill for small blogs.  I could also enable `ARCHIVES_ARE_INDEXES` and `CREATE_DAILY_ARCHIVES` for a better effect.
3. Still, your plugin should check that it doesn’t conflict with archives.  Otherwise, a nice error will prevent the site from building due to *two tasks having the same target*.

**Comment by bwinton** [comment]
Ah!  Yes!  I should definitely make it check for that.
Is there a clean way to say "Does this target already exist?" (or "Is another plugin going to handle this target?") so that I can skip generating it?  (That would let me avoid the check at https://github.com/bwinton/plugins/blob/master/v7/subindexes/subindexes.py#L58 too!)

**Comment by Kwpolska** [comment]
No, you’d have to check possibilities manually, according to current archive settings.

**Pull request merged**